docs/user-guide/11-target-specifics.md

```diff
@@ -0,0 +1,144 @@
+
+- [[vk::input_attachment_index(i)]]
+    Same as `input_attachment_input` layout qualifier in Vulkan. It selects which subpass input is being read from. It is valid only when used on subpassInput type uniform variables.
+
+- [[vk::push_constant]]
+    Same as `push_constant` layout qualifier in Vulkan. It is applicable only to a uniform block and it will be copied to a special memory location where GPU may have a more direct access to.
+
+- [vk::image_format(format : String)]
+    Same as `[[vk::image_format("XX")]]` layout qualifier in DXC. Vulkan/GLSL allows the format string to be specified without the keyword, `image_format`.  Consider the following Slang code, as an example,
+  ```csharp
+  [vk::image_format("r32f")] RWTexture2D<float> typicalTexture;
+  ```
+  It will generate the following GLSL or SPIR-V code.
+  > layout(r32f) uniform image2D typicalTexture_0;
+  > %18 = OpTypeImage %float 2D 2 0 0 2 R32f
+
+- [vk::shader_record]
+    Same as `shaderRecordEXT` layout qualifier in [GL_EXT_ray_tracing extension](https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_ray_tracing.txt).
+    It can be used on a buffer block that represents a buffer within a shader record as defined in the Ray Tracing API.
+
+- [vk::aliased_pointer]
+    TODO
+
+- [vk_restrict_pointer]
+    TODO
+
+- [vk::spirv_instruction(op : int, set : String = "")]
+    It is similar to how `__intrinsic_op` works, but it uses the `op-code` of SPIR-V instruction set. You can also specify which instruction set you want to use as a second argument. E.G., `[[vk::spirv_instruction(1, "NonSemantic.DebugBreak")]]`
+
+- [spv_target_env_1_3]
+    TODO: I am not sure what it does. There isn't an example on slang-test
+
+
+Target specific Slang intrinsics
+--------------------------------
+
+### SPIR-V specific keywords and intrinsics
+
+Slang doesn't support the following Precision qualifiers in Vulkan. They are simply ignored and all are treated as highp.
+ - lowp : RelaxedPrecision, on storage variable and operation
+ - mediump : RelaxedPrecision, on storage variable and operation
+ - highp : 32-bit, same as int or float
+
+TODO: Cannot think of anything to put here. `spirv_asm` is already well documented in a1-04-interop.md.
+
+
+Memory pointer
+--------------
+
+TODO: Describe any information when using memory pointers for SPIR-V and Metal
+
+
+Buffer layout and alignment
+---------------------------
+
+TODO: Need to digest the comments writting in "compute/buffer-layout.slang" and write up a user-friendly document here.
+
+
+Matrix majorness (row/column)
+-----------------------------
+
+### Matrix majorness for SPIR-V
+
+TODO: Not sure how this is handled.
```

Comment:
I think we just want to say that we are following the same behavior in DXC to translate a N-row-by-M-column matrix in Slang maps to a N-column-by-M-row matrix in GLSL/SPIRV. If you can talk more about the reasoning behind this decision, that would be great.